nvflare/lighter/utils.py

```diff
@@ -181,12 +181,42 @@ def sign_all(content_folder, signing_pri_key):
 
 
 def load_yaml(file):
+
+    root = os.path.split(file)[0]
+    yaml_data = None
     if isinstance(file, str):
-        return yaml.safe_load(open(file, "r"))
+        yaml_data = yaml.safe_load(open(file, "r"))
```

Comment:
This will leave the file open forever. Please change to:
with open(file, "r") as f:
  yaml_data = yaml.safe_load(f).

There are probably other places like this.

Reply:
Changed. Also changes all other places with the same open file pattern.